homeassistant/components/group/__init__.py

```diff
@@ -293,14 +293,36 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     await _async_process_config(hass, config)
 
     async def reload_service_handler(service: ServiceCall) -> None:
-        """Remove all user-defined groups and load new ones from config."""
-        auto = [e for e in component.entities if e.created_by_service]
+        """Group reload handler.
 
-        if (conf := await component.async_prepare_reload()) is None:
+        - Remove group.group entities not created by service calls and set them up again
+        - Reload xxx.group platforms
+        """
+        if (conf := await component.async_prepare_reload(skip_reset=True)) is None:
             return
-        await _async_process_config(hass, conf)
 
-        await component.async_add_entities(auto)
+        # There's only a group.group platform, no group.xxx platforms with
+        # entities.
+        # pylint: disable-next=protected-access
+        platform = component._platforms[DOMAIN]
+
+        # Simplified + modified version of EntityPlatform.async_reset:
+        # - group.group never retries setup
+        # - group.group never polls
+        # - We don't need to reset EntityPlatform._setup_complete
+        # - Only remove entities which were not created by service calls
+        tasks = [
+            entity.async_remove()
+            for entity in platform.entities.values()
+            if not cast(Group, entity).created_by_service
+        ]
```

Comment:
```suggestion
        # There's only a group.group platform, no group.xxx platforms with
        # Simplified + modified version of EntityPlatform.async_reset:
        # - group.group never retries setup
        # - group.group never polls
        # - We don't need to reset EntityPlatform._setup_complete
        # - Only remove entities which were not created by service calls
        tasks = [
            entity.async_remove()
            for entity in component.entities
            if entity.entity_id.startswith("group.")
            and not cast(Group, entity).created_by_service
        ]
```

Maybe avoid the protected access?